### PR TITLE
make many-to-one relations append __id

### PR DESCRIPTION
**Pull request by cameronfromtorq**
currently this will just append _)id to manytoone field. 

If its worthwhile, I can try to find a way for this to be hidden from the user: 
image

### CODE REVIEW

**Comment by cameronfromtorq** [comment]
looking into handling this in the backend quickly, if not also add xxxx__type as an option

**Review by lukemacausland — CHANGES_REQUESTED** [review]
I like this pattern. One comment I have is that once saved, we are able to remove one of these fields, and than re-add the ManyToOneRelation field which makes it throw an error.
image

File: models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -1422,6 +1422,15 @@ public function getCompositeIndices(): array
      */
     public function setCompositeIndices($compositeIndices)
     {
+        $class = $this->getFieldDefinitions([]);
+        foreach($compositeIndices as $indexInd => $compositeIndex){
+            foreach($compositeIndex["index_columns"] as $fieldInd => $fieldName){
+                if(isset($class[$fieldName]) && str_contains(strtoupper($class[$fieldName]->getFieldType()), "MANYTOONE")){
```
Comment (lukemacausland) [inline models/DataObject/ClassDefinition.php]:
```suggestion
                if(isset($class[$fieldName]) && $class[$fieldName] instanceof ManyToOneRelation){
```

Comment (lukemacausland) [reply models/DataObject/ClassDefinition.php]: I would suggest writing it like this, than we aren't hard coding a field type name. Will also need to add the ManyToOneRelation import at the top.

`use Pimcore\Model\DataObject\ClassDefinition\Data\ManyToOneRelation;`

Comment (cameronfromtorq) [reply models/DataObject/ClassDefinition.php]: > I like this pattern. One comment I have is that once saved, we are able to remove one of these fields, and than re-add the ManyToOneRelation field which makes it throw an error. image

I would think doing a array_unique check on the back end would fix this issue but im, getting different issues after doing that. The backend seems to like a certain order for the indicies that seems random to me.

**Comment by cameronfromtorq** [comment]
I like this pattern. One comment I have is that once saved, we are able to remove one of these fields, and than re-add the ManyToOneRelation field which makes it throw an error.

I would think doing a array_unique check on the back end would fix this issue but im, getting different issues after doing that. The backend seems to like a certain order for the indicies that seems random to me.

**Comment by lukemacausland** [comment]
@cameronfromtorq are we good to close this since you have https://github.com/TorqIT/pimcore/pull/35 open?

**Comment by cameronfromtorq** [comment]
yes!

**Pull request closed without merging**